io.openems.edge.io.shelly/src/io/openems/edge/io/shelly/shellyplug/IoShellyPlugImpl.java

```diff
@@ -64,9 +70,21 @@ public IoShellyPlugImpl() {
 	@Activate
 	private void activate(ComponentContext context, Config config) {
 		super.activate(context, config.id(), config.alias(), config.enabled());
-		this.shellyApi = new ShellyApi(config.ip());
 		this.meterType = config.type();
 		this.phase = config.phase();
+		this.baseUrl = "http://" + config.ip();
+
+		if (!this.isEnabled()) {
+			return;
+		}
+		this.httpBridge.subscribeJsonEveryCycle(this.baseUrl + "/status", t -> {
+			this.processHttpResult(t);
+			this._setSlaveCommunicationFailed(false);
+		}, t -> {
+			this.logDebug(this.log, t.getMessage());
+			this._setRelay(null);
```

Comment:
The problem here is, that the 'Relay' Channel gets set properly, but you forgot `ActivePower` and `ActiveProductionEnergy`. This is a very typical mistake, but will lead to wrong data, e.g. in the UI.

My approach would be to use one common `BiConsumer` that handles the result and a exception. Also `processHttpResult` could then take both and act accordingly. This way you set all Channels in one place and it's more likely to not forget anything.